### PR TITLE
Whitelist users don't build them.

### PR DESCRIPTION
**Pull request by mmcdermott**
This will necessitate changes in the add user/delete user form, as it changes the returned value from the API in the invitation case vs. the membership case.

### CODE REVIEW

File: spec/mailers/previews/user_organization_invitation_mailer_preview.rb [hunk 1]
```diff
@@ -7,4 +11,5 @@ def new_user_invite
     redirect_url = "fake"
     UserOrganizationMembershipMailer.new_user_invite user, organization, redirect_url, password
   end
+>>>>>>> master:spec/mailers/previews/user_organization_membership_mailer_preview.rb
```
Comment (OAGr) [inline spec/mailers/previews/user_organization_invitation_mailer_preview.rb]: conflict; fix this

File: app/models/user_organization_invitation.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class UserOrganizationInvitation < ActiveRecord::Base
+  belongs_to :organization
+
+  validates_presence_of :organization
+  validates :email, presence: true, format: /@/
```
Comment (OAGr) [inline app/models/user_organization_invitation.rb]: what is the /@/ format?  Is that a regex to make sure it has an @ in it?

Comment (mmcdermott) [inline app/models/user_organization_invitation.rb]: Yes.

File: app/controllers/user_organization_memberships_controller.rb [hunk 3]
```diff
@@ -18,21 +16,11 @@ def organization_memberships
...
-    end
-
-    @membership = UserOrganizationMembership.new user: @user, organization: @organization
-    if @membership.save
-      InternalMailer.organization_changed_member_count(@organization, 'added').deliver_later
-      render json: OrganizationMembershipRepresenter.new(@membership).to_json
+      invite_new_user
     else
-      render json: @membership.errors, status: :unprocessable_entity
+      create_new_membership
```
Comment (OAGr) [inline app/controllers/user_organization_memberships_controller.rb]: If I'm reading this correctly, then this returns either a OrganizationMembershipRepresentor or a UserOrganizationInvitationRepresentor.  That's confusing on the front end.

**Pull request merged**